test/invariant/fuzz/handlers/HandlerAllo.t.sol

```diff
@@ -33,65 +34,299 @@ contract HandlerAllo is Setup {
 
-    function handler_updatePoolMetadata(
-        uint256 _idSeed,
-        uint256 _metadataProtocol,
-        string calldata _data
-    ) public {
+    function handler_updatePoolMetadata(uint256 _idSeed, uint256 _metadataProtocol, string calldata _data) public {
         // Needs at least one pool
         if (ghost_poolIds.length == 0) return;
 
         _idSeed = _idSeed % ghost_poolIds.length;
         uint256 poolId = ghost_poolIds[_idSeed];
 
         // Get the profile ID
-        IRegistry.Profile memory profile = registry.getProfileByAnchor(
-            _ghost_anchorOf[msg.sender]
-        );
+        IRegistry.Profile memory profile = registry.getProfileByAnchor(_ghost_anchorOf[msg.sender]);
 
         // Avoid EOA
         if (profile.anchor == address(0)) return;
 
-        Metadata memory metadata = Metadata({
-            protocol: _metadataProtocol,
-            pointer: _data
-        });
+        Metadata memory metadata = Metadata({protocol: _metadataProtocol, pointer: _data});
 
         // Update the pool metadata - will revert on wrong anchor
-        targetCall(
+        targetCall(address(allo), 0, abi.encodeWithSelector(IAllo.updatePoolMetadata.selector, poolId, metadata));
+    }
+
+    function handler_updatePercentFee(uint256 _newPercentFee) public {
+        _newPercentFee = bound(_newPercentFee, 0, 1e18);
+
+        // Update the percent fee - will revert if caller is not the owner
+        targetCall(address(allo), 0, abi.encodeWithSelector(IAllo.updatePercentFee.selector, _newPercentFee));
+    }
+
+    function handler_updateBaseFee(uint256 _newBaseFee) public {
+        // Update the base fee - will revert if caller is not the owner
+        targetCall(address(allo), 0, abi.encodeWithSelector(IAllo.updateBaseFee.selector, _newBaseFee));
+    }
+
+    function handler_updateRegistry(address _newRegistry) public {
+        // Update the registry - will revert if caller is not the owner or if the new registry is zero
+        targetCall(address(allo), 0, abi.encodeWithSelector(IAllo.updateRegistry.selector, _newRegistry));
+    }
+
+    function handler_updateTreasury(address _newTreasury) public {
+        // Update the treasury - will revert if caller is not the owner or if the new treasury is zero
+        targetCall(address(allo), 0, abi.encodeWithSelector(IAllo.updateTreasury.selector, _newTreasury));
+    }
+
+    function handler_updateTrustedForwarder(address _newForwarder) public {
+        // Update the trusted forwarder - will revert if caller is not the owner or if the new forwarder is zero
+        targetCall(address(allo), 0, abi.encodeWithSelector(Allo.updateTrustedForwarder.selector, _newForwarder));
+    }
+
+    function handler_addPoolManagers(uint256 _idSeed, address[] calldata _managers)
+        public
+        onlyValidPoolAndAdmin(_idSeed)
+    {
+        uint256 _poolId = _getPoolId(_idSeed);
+
+        // Add pool managers - will revert if caller is not the pool admin of the pool id
```

Comment:
with the modifier, this will never revert (for these reasons)?
I'd be careful to not add too many constraints (yet?) -> it's ok to let it revert for these reasons (esp as potentially, this could/should be tested as a property in the future -> with the sender/pool id constrained, test would only cover the happy path)

Did you add these constraints to increase coverage? I guess if all the calls were reverting due to a wrong caller, and as we already reduce the caller space (via actors), it "just" needs more run, no?

Reply:
Yes, I can try removing the constraints one by one. Also, if we never check why something reverts in the handlers, we do that in the properties or the unit tests. Is it best for the handlers to have the max fuzz input to generate the state changing with the least number of reverts? Since we never validate those?